static/app/views/dashboardsV2/widgetCard/releaseWidgetQueries.tsx

```diff
@@ -377,87 +343,22 @@ class ReleaseWidgetQueries extends Component<Props, State> {
-      widget.queries[0].orderby,
-      useSessionAPI
-    );
-    const columns = widget.queries[0].columns;
-
-    const includeSeries = widget.displayType !== DisplayType.TABLE ? 1 : 0;
-    const includeTotals =
-      widget.displayType === DisplayType.TABLE ||
-      widget.displayType === DisplayType.BIG_NUMBER ||
-      columns.length > 0
-        ? 1
-        : 0;
-
-    widget.queries.forEach(query => {
-      let requestData;
-      let requester;
-      if (useSessionAPI) {
-        const sessionAggregates = aggregates.filter(
-          agg =>
-            !!!Object.values(DerivedStatusFields).includes(agg as DerivedStatusFields)
-        );
-        requestData = {
-          field: sessionAggregates,
-          orgSlug: organization.slug,
-          end,
-          environment: environments,
-          groupBy: columns,
-          limit: undefined,
-          orderBy: '', // Orderby not supported with session.status
-          interval,
-          project: projects,
-          query: query.conditions,
-          start,
-          statsPeriod: period,
-          includeAllArgs,
-          cursor,
-        };
-        requester = doSessionsRequest;
-      } else {
-        requestData = {
-          field: aggregates.map(fieldsToDerivedMetrics),
-          orgSlug: organization.slug,
-          end,
-          environment: environments,
-          groupBy: columns.map(fieldsToDerivedMetrics),
-          limit: columns.length === 0 ? 1 : isCustomReleaseSorting ? 100 : this.limit,
-          orderBy: unsupportedOrderby
-            ? ''
-            : isDescending
-            ? `-${fieldsToDerivedMetrics(rawOrderby)}`
-            : fieldsToDerivedMetrics(rawOrderby),
-          interval,
-          project: projects,
-          query:
-            query.conditions + (releaseCondition === '' ? '' : ` ${releaseCondition}`),
-          start,
-          statsPeriod: period,
-          includeAllArgs,
-          cursor,
-          includeSeries,
-          includeTotals,
-        };
-        requester = doMetricsRequest;
-
-        if (
-          rawOrderby &&
-          !!!unsupportedOrderby &&
-          !!!aggregates.includes(rawOrderby) &&
-          !!!columns.includes(rawOrderby)
-        ) {
-          requestData.field = [...requestData.field, fieldsToDerivedMetrics(rawOrderby)];
-          if (!!!injectedFields.includes(rawOrderby)) {
-            injectedFields.push(rawOrderby);
-          }
-        }
-      }
+    if (!useSessionAPI) {
+      widget.queries.forEach(query => {
+        query.conditions =
+          query.conditions + (releaseCondition === '' ? '' : ` ${releaseCondition}`);
+      });
+    }
 
-      promises.push(requester(api, requestData));
-    });
+    const promises: Promise<
+      MetricsApiResponse | [MetricsApiResponse, string, ResponseMeta] | SessionApiResponse
+    >[] = widget.queries.map(query =>
+      this.config.getTableRequest!(widget, query, this.limit, cursor, {
```

Comment:
Since we're going to split the config into 'getTableRequests' and 'getSeriesRequests', releaseWidgetQueries should also follow the same pattern. but all the common parts can be abstracted into a common function and `getTableRequests` and `getSeriesRequests` can be resposible for setting the `includeSeries` and `includeTotals` value?